src/types/timeline.ts

```diff
@@ -1,34 +1,46 @@
 import type { Selection } from 'd3-selection';
-import type { ActivityDirective, ActivityType } from './activity';
+import type { ActivityDirective, ActivityDirectiveId, ActivityType } from './activity';
 import type { ConstraintResultWithName } from './constraint';
-import type { ResourceType, Span } from './simulation';
+import type { ExternalEvent, ExternalEventId, ExternalEventType } from './external-event';
+import type { ResourceType, Span, SpanId } from './simulation';
 
-export type ActivityTree = ActivityTreeNode[];
+export type DiscreteTree = DiscreteTreeNode[];
 
-export type ActivityTreeNode = {
-  children: ActivityTreeNode[]; // Child nodes of this directive/span/aggregation
+export type DiscreteTreeNode = {
+  activity_type: undefined | 'aggregation' | 'directive' | 'span'; // when items are activity related, this applies. TODO: this design is a little awkward.
+  children: DiscreteTreeNode[];
   expanded: boolean;
   id: string;
   isLeaf: boolean;
-  items: ActivityTreeNodeItem[]; // Directives and/or spans to render on the timeline at this level of the tree
+  items: DiscreteTreeNodeItem[];
   label: string;
-  type: 'aggregation' | 'directive' | 'span';
+  type: 'Activity' | 'ExternalEvent';
 };
 
-export type ActivityTreeNodeItem = { directive?: ActivityDirective; span?: Span };
+// export type DiscreteTreeNodeItem = { externalEvent?: ExternalEvent } | { directive?: ActivityDirective; span?: Span };
+export type DiscreteTreeNodeItem = { directive?: ActivityDirective; externalEvent?: ExternalEvent; span?: Span };
 
-export type ActivityTreeNodeDrawItem = ActivityTreeNodeItem & { startX: number };
+export type DiscreteTreeNodeDrawItem = DiscreteTreeNodeItem & { startX: number };
 
-export type ActivityTreeExpansionMap = Record<string, boolean>;
+// export type ActivityTreeExpansionMap = Record<string, boolean>;
+// export type ExternalEventTreeExpansionMap = Record<string, boolean>;
+export type DiscreteTreeExpansionMap = Record<string, boolean>;
 
 export interface ActivityLayer extends Layer {
   activityColor: string;
   activityHeight: number; // @deprecated TODO how should we deprecate view properties?
 }
+export interface ExternalEventLayer extends Layer {
+  externalEventColor: string;
+  externalEventHeight: number; // @deprecated TODO how should we deprecate view properties?
```

Comment:
Remove this? Since this code hasn't been released we don't need to keep this around. That said I did leave another comment about view property deprecation/change/migration.

Reply:
I believe we replicated that comment to match `activityHeight`'s comment - should we remove it for both?

Reply:
Aha right, forgot that we did that. We need to find a good mechanism to deprecate activityHeight but haven't removed it yet since it has been out there for a while and exists in current views. `externalEventHeight` has not been released so doesn't need to be added since we have no need for it internally.

Reply:
Removed from `types/timeline.ts` and the view schema!